### ISSUE TITLE
OUser.hasRole() does not work correctly  in case of child ORole class

### ISSUE THREAD

**Issue by PhantomYdn** [orientechnologies/orientdb#8338]
### OrientDB Version: 2.2.30, 3.0.3

There is such line in code: https://github.com/orientechnologies/orientdb/blob/develop/core/src/main/java/com/orientechnologies/orient/core/metadata/security/OSecurityShared.java#L319

```java
  public ORole getRole(final OIdentifiable iRole) {
    final ODocument doc = iRole.getRecord();
    if (doc != null && "ORole".equals(doc.getClassName()))
      return new ORole(doc);

    return null;
  }
```

As you can see, there is explicit check for className - it should be equal ORole. But if you extend ORole and use for your project child class: it will lead to bug when hasRole doesn't work.

**Comment by luigidellaquila** [comment]
Hi @PhantomYdn 

Thank you for reporting, I'm fixing it now

Thanks

Luigi

**Comment by luigidellaquila** [comment]
Hi @PhantomYdn 

I just pushed a fix, now it should be OK.
The fix will be released with v 2.2.37 and v 3.0.3

Thanks

Luigi
